model/textparse/protobufparse_test.go

```diff
@@ -3214,3 +3221,184 @@ func TestProtobufParse(t *testing.T) {
 		})
 	}
 }
+
+func FuzzProtobufParser_Labels(f *testing.F) {
+	f.Add(true, true, int64(123))
+	f.Add(true, false, int64(129))
+	f.Add(false, true, int64(159))
+	f.Add(false, true, int64(-127))
+	f.Fuzz(func(
+		t *testing.T,
+		parseClassicHistogram bool,
+		enableTypeAndUnitLabels bool,
+		randSeed int64,
+	) {
+		var (
+			r              = rand.New(rand.NewSource(randSeed))
```

Comment:
This was surprising to me.
The [docs](https://pkg.go.dev/testing@master#F.Fuzz) suggest "This function should be fast and deterministic".

Reply:
I'm not particularly against it, but I do think it should be clear what is going on.

This would be the first test using `testing.F` in Prometheus.
We have a [fuzzing step in CI](https://github.com/prometheus/prometheus/actions/runs/16749354339/job/47415408837?pr=16957), but it uses a different pattern.  It's not running `FuzzProtobufParser_Labels`.

AFAICS during normal `go test`, and in CI, `FuzzProtobufParser_Labels` runs exactly the 4 cases you have added to the corpus.

@roidelapluie you created the existing fuzz tests, I think.  Do you have an opinion about this one?

Reply:
Thanks for pointing out that `testing.F` is not run in the fuzzing step in CI. At this point I feel the PR is a bit stalled by this decision, would be great to pick one of the following options:
1. Turn this test back into normal fuzzing test
2. Enable Go fuzzing in fuzzing step in CI

Or something else, I'm open.